simulators/kcc/sync/main.go

```diff
@@ -0,0 +1,163 @@
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/ethereum/hive/hivesim"
+)
+
+var (
+	// the number of seconds before a sync is considered stalled or failed
+	syncTimeout = 65 * time.Second
+	params      = hivesim.Params{
+		"HIVE_CLIQUE_PRIVATEKEY": "9c647b8b7c4e7c3490668fb6c11473619db80c93704c70893d3813af4090c39c",
+		"HIVE_MINER":             "658bdf435d810c91414ec09147daa6db62406379",
+		"HIVE_NETWORK_ID":         "321",
+		"HIVE_CHAIN_ID":       "321",
+
+		"HIVE_NODETYPE":       "",
+		// block interval: 1s
+		"HIVE_KCC_POSA_BLOCK_INTERVAL": "1",
+		// epoch : 5
+		"HIVE_KCC_POSA_EPOCH": "10",
+		// initial valiators
+		"HIVE_KCC_POSA_ISHIKARI_INIT_VALIDATORS": "0x658bdf435d810c91414ec09147daa6db62406379",
+		// admin
+		"HIVE_KCC_POSA_ADMIN": "0x658bdf435d810c91414ec09147daa6db62406379",
+		// KCC Ishikari  fork number
+		"HIVE_FORK_KCC_ISHIKARI": "9",
+		// KCC Ishikari Patch001 fork number
+		"HIVE_FORK_KCC_ISHIKARI_PATCH001": "10",
+		// KCC Ishikari Patch002 fork number
+		"HIVE_FORK_KCC_ISHIKARI_PATCH002": "11",
+	}
+	sourceFiles = map[string]string{
+		"genesis.json": "./simplechain/genesis.json",
+		"chain.rlp":    "./simplechain/chain.rlp",
+		"geth.sh": 		"./simplechain/geth.sh",
+	}
+	sinkFiles = map[string]string{
+		"genesis.json": "./simplechain/genesis.json",
+		"geth.sh": "./simplechain/geth.sh",
+	}
+)
+
+var (
+	testchainHeadNumber = uint64(8591)
+	testchainHeadHash   = common.HexToHash("0x0520a5c756a2b225ac8e35fbd027f82c3fb21082a1f53c178540df5e94000013")
+)
+
+func main() {
+	var suite = hivesim.Suite{
+		Name: "sync with other clients",
+		Description: `This suite of tests verifies that clients can sync from each other in different modes.
+For each client, we test if it can serve as a sync source for all other clients (including itself).`,
+	}
+	suite.Add(hivesim.ClientTestSpec{
+		Role:        "eth1",
+		Name:        "CLIENT as sync source",
+		Description: "This loads the test chain into the client and verifies whether it was imported correctly.",
+		Parameters:  params,
+		Files:       sourceFiles,
+		Run:         runSourceTest,
+	})
+	hivesim.MustRunSuite(hivesim.New(), suite)
+}
+
+func runSourceTest(t *hivesim.T, c *hivesim.Client) {
+	// Check whether the source has imported its chain.rlp correctly.
+	source := &node{c}
+	if err := source.checkHead(testchainHeadNumber, testchainHeadHash); err != nil {
+		t.Fatal(err)
+	}
+
+	// Configure sink to connect to the source node.
+	enode, err := source.EnodeURL()
+	if err != nil {
+		t.Fatal("can't get node peer-to-peer endpoint:", enode)
+	}
+	t.Logf("enode: %s\n", enode)
+	sinkParams := params.Set("HIVE_BOOTNODE", enode)
+
+	// Sync all sink nodes against the source.
+	t.RunAllClients(hivesim.ClientTestSpec{
+		Role:        "eth1",
+		Name:        fmt.Sprintf("sync %s -> CLIENT", source.Type),
+		Description: fmt.Sprintf("This test attempts to sync the chain from a %s node.", source.Type),
+		Parameters:  sinkParams,
```

Comment:
Looks like the two nodes use the same account as their miner account.  Does this cause chaos?